amazon-msk/cdk/lib/WebStreamingStack.ts

```diff
@@ -0,0 +1,431 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as secretsmanager from 'aws-cdk-lib/aws-secretsmanager';
+import * as logs from 'aws-cdk-lib/aws-logs';
+import { Construct } from 'constructs';
+import * as path from 'path';
+import Mustache = require("mustache");
+import fs =  require("fs");
+
+interface TemplateData {
+  name: string;
+  vault: string;
+  glue?: object;
+  cloudwatch?: object;
+  mappings?: Array<object>;
+  public?: object;
+  kafka?: object;
+  jwt?: object;
+}
+
+interface WebStreamingMskContext {
+  servers: string,
+  credentials: WebStreamingMskCredentialsContext
+}
+
+interface WebStreamingMskCredentialsContext {
+  sasl: string
+}
+
+interface WebStreamingJwtContext {
+  issuer: string,
+  audience: string,
+  keysUrl: string
+}
+
+interface WebStreamingGlueContext {
+  registry: string
+}
+
+interface WebStreamingPublicContext {
+  servers: string,
+  certificate: string
+}
+
+interface WebStreamingMappingContext {
+  topic: string,
+  automatic: boolean,
+  path: string,
+}
+
+interface WebStreamingCloudWatchContext {
+  metrics?: WebStreamingCloudWatchMetricsContext,
+  logs?: WebStreamingCloudWatchLogsContext
+}
+
+interface WebStreamingCloudWatchMetricsContext {
+  namespace: string
+}
+
+interface WebStreamingCloudWatchLogsContext {
+  group: string,
+  stream?: string
+}
+
+interface WebStreamingContext {
+  vpcId: string,
+  subnetIds: Array<string>,
+  msk: WebStreamingMskContext;
+  public: WebStreamingPublicContext;
+  mappings: Array<WebStreamingMappingContext>,
+  jwt: WebStreamingJwtContext,
+  glue: WebStreamingGlueContext,
+  cloudwatch?: WebStreamingCloudWatchContext,
+  securityGroup?: string,
+  roleName?: string,
+  capacity?: number,
+  instanceType?: string,
+  sshKey?: string,
+  ami?: string,
+  version?: string
+}
+
+export class WebStreamingStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // lookup context
+    const context = this.node.getContext(id) as WebStreamingContext;
+
+    // detect dependencies
+    const nitroEnclavesEnabled: boolean = context.public.certificate.startsWith("arn:aws:acm");
+    const secretsmanagerEnabled: boolean = context.public.certificate.startsWith("arn:aws:secretsmanager");
+    const cloudwatchEnabled: boolean =
+      context.cloudwatch?.logs?.group !== undefined ||
+      context.cloudwatch?.metrics?.namespace !== undefined;
+
+    // apply context defaults
+    context.version ??= "25.4.4"; // TODO "latest" (currently unresolveable)
+    context.capacity ??= 2;
+    context.instanceType ??= nitroEnclavesEnabled ? 'c6i.xlarge' : 't3.small';
+    context.mappings ??= [];
+    context.mappings.forEach((mapping: WebStreamingMappingContext) => {
+      mapping.automatic ??= true;
+      mapping.path ??= `/${mapping.topic}`;
+    });
+
+    const [externalServer, externalPort] = context.public.servers.split(',')[0].split(':');
```

Comment:
This is not correct with the current format of `https://your.custom.domain:7143/` in the examples. As there is 2 `:` characters. Also applies to IOT, but there we have: `<your public wildcard servers including port>`. So I'm not sure what kind of format should be the input here.